spec/preservation_ingest/complete_ingest_spec.rb

```diff
@@ -3,13 +3,15 @@
   let(:druid) { "druid:#{bare_druid}" }
   let(:deposit_dir_pathname) { Pathname(File.join(File.dirname(__FILE__), '..', 'fixtures', 'deposit', 'complete-ingest')) }
   let(:deposit_bag_pathname) { Pathname(File.join(deposit_dir_pathname, bare_druid)) }
-  let(:mock_so) { instance_double(Moab::StorageObject, deposit_bag_pathname: deposit_bag_pathname) }
+  let(:mock_storage_object) { instance_double(Moab::StorageObject, deposit_bag_pathname: deposit_bag_pathname) }
+  let(:mock_storage_objects) { [mock_storage_object] }
   let(:this_robot) { described_class.new }
 
   describe '#perform' do
     before do
       allow(deposit_bag_pathname).to receive(:rmtree)
-      allow(Moab::StorageServices).to receive(:find_storage_object).and_return(mock_so)
+      allow(Moab::StorageServices).to receive(:search_storage_objects).and_return(mock_storage_objects)
+      allow(mock_storage_objects).to receive(:filter!).and_return(mock_storage_object)
```

Comment:
i think we should prob test one level below the `filter!` call, instead mocking `return` and `raise` behavior from `Preservation::Client.objects.primary_moab_location`, `Stanford::StorageServices.search_storage_objects`, and `Stanford::StorageServices.find_storage_object`, so that we simulate the storage interactions a bit more thoroughly and exercise the `moab_object` changes a bit more thoroughly.